OpenSim/Sandbox/MatlabScripts/osimTableToStruct.m

```diff
@@ -0,0 +1,91 @@
+% for more information. OpenSim is developed at Stanford University       %
+% and supported by the US National Institutes of Health (U54 GM072970,    %
+% R24 HD065690) and by DARPA through the Warrior Web program.             %
+%                                                                         %
+% Copyright (c) 2005-2012 Stanford University and the Authors             %
+% Author(s): James Dunne                                                  %
+%                                                                         %
+% Licensed under the Apache License, Version 2.0 (the "License");         %
+% you may not use this file except in compliance with the License.        %
+% You may obtain a copy of the License at                                 %
+% http://www.apache.org/licenses/LICENSE-2.0.                             %
+%                                                                         %
+% Unless required by applicable law or agreed to in writing, software     %
+% distributed under the License is distributed on an "AS IS" BASIS,       %
+% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or         %
+% implied. See the License for the specific language governing            %
+% permissions and limitations under the License.                          %
+% ----------------------------------------------------------------------- %
+
+% Author: James Dunne, Shrinidhi K. Lakshmikanth, Chris Dembia, Tom Uchida,
+% Ajay Seth, Ayman Habib, Jen Hicks.
+
+%%
+import org.opensim.modeling.*
+
+%%
+nCol = table.getNumColumns;
+nRow = table.getNumRows;
+
+%% make an empty data structure
+data = struct();
+
+%%
+if strcmp( char(table.getClass), 'class org.opensim.modeling.TimeSeriesTableVec3')
+    rowdata = zeros(nRow, 3);
+elseif strcmp( char(table.getClass), 'class org.opensim.modeling.TimeSeriesTable')
+    rowdata = zeros(nRow, 1);
+else
+    disp(['Input is of type ' char(table.getClass)'])
+    error(['This function only deals with TimeSeriesTable or TimeSeriesTableVec3']);
+end
+
+%%
+for iCol = 0 : nCol -1
+
+    if strcmp( char(table.getClass), 'class org.opensim.modeling.TimeSeriesTableVec3')
+
+        for iRow = 0 : nRow - 1
+            rowdata(iRow+1,1) = table.getDependentColumnAtIndex(iCol).getElt(iRow,1).get(0);
+            rowdata(iRow+1,2) = table.getDependentColumnAtIndex(iCol).getElt(iRow,1).get(1);
+            rowdata(iRow+1,3) = table.getDependentColumnAtIndex(iCol).getElt(iRow,1).get(2);
+        end
+
+    elseif strcmp( char(table.getClass), 'class org.opensim.modeling.TimeSeriesTable')
+
+        for iRow = 0 : nRow - 1
+            rowdata(iRow+1,1) = table.getDependentColumnAtIndex(iCol).getElt(iRow,0);
+        end
+    end
+
+    col_label  = char(table.getColumnLabels.get(iCol));
+
+    if isempty(strfind(col_label, '/'))
+        eval(['[data.' col_label '] = rowdata;']);
+    else
+       temp = strfind(col_label, '/');
+       new_col_label = col_label(temp(end-1)+1:temp(end)-1);
```

Comment:
If `col_label` is `a/b/c`, it looks like you're just using `b` as the field name. That strategy won't work in general since field names must be unique and there could be more than one `c` (Output) under each `b` (Component). Perhaps replace `/` with `_`?